### PR TITLE
Add xy_bounds keyword for PSF photometry

### LINKED ISSUE #1801: Limit offsets when fitting
As noted in #1799, it is possible to limit the flux to be positive by setting the model flux min to be zero:
```python
    dao_psf_model.flux.min = 0
```

which is useful for rejecting negative sources during fitting.

I'd like a mechanism to do the same for `x_0` and `y_0`, but restricting those parameters doesn't work since they need to be set to the local x,y value.  Instead, the limits need to be set independently for each source.  I think that could be accommodated in `make_psf_model`.

One use case is very crowded fields:  I've seen DAOFind-found peaks close enough that they jump within the 5x5 fit zone onto another star's peak.  

A workaround I'll try (but expect to have limited success with) is grouping.

### PR DESCRIPTION
**Pull request by larrybradley**
This PR adds a new ``xy_bounds`` keyword to ``PSFPhotometry`` and ``IterativePSFPhotometry`` to allow one to bound the x and y model parameters during the fitting.  If any fitted x or y value is at the bounded value, then a new flag value of 32 is added to the bitwise `flags` column.

Closes #1801 

CC: @keflavich

### CODE REVIEW

File: photutils/psf/photometry.py [hunk 1]
```diff
@@ -289,20 +297,21 @@ class for more information.
...
                  fitter=LevMarLSQFitter(), fitter_maxiters=100,
-                 localbkg_estimator=None, aperture_radius=None,
+                 xy_bounds=None, localbkg_estimator=None, aperture_radius=None,
                  progress_bar=False):
 
         self._param_maps = self._define_param_maps(psf_model)
         self.psf_model = _validate_psf_model(psf_model)
 
-        self.fit_shape = as_pair('fit_shape', fit_shape, lower_bound=(0, 1),
+        self.fit_shape = as_pair('fit_shape', fit_shape, lower_bound=(1, 0),
```
Comment (keflavich) [inline photutils/psf/photometry.py:306]: Is this change intentional?  I'm not sure what it does, but it seems smuggled in

**Review by keflavich — APPROVED** [review]

File: photutils/psf/photometry.py (hunk 1 above)
Comment (larrybradley) [reply photutils/psf/photometry.py:306]: It's a tiny "fix" that I noticed during testing.  All this change does is change the error message if one were to input `fit_shape = 0`.  It could be a separate PR, but it's tiny.....
